projects/lights_control/src/lights_can.c

```diff
@@ -0,0 +1,109 @@
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+
+#include "can_unpack_impl.h"
+#include "event_queue.h"
+#include "status.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+#include "lights_gpio.h"
+
+#define SOMEPORT 0
+#define SOMEPIN 0
+
+#define CASE_ACTION(action)            \
```

Comment:
I'd rather avoid complicating a simple switch case with all this Macro boilerplate. While it is more efficient to type it is much clearer to other what is happening if you don't do this. It is a tradeoff of readability vs efficiency and we tend to favor readability.

Reply:
I used ControlTables as @karlding  suggested. My code looks much more sexy now 💯